src/modules/deal/deal.service.test.ts

```diff
@@ -0,0 +1,318 @@
+  const valueGenerator = new RandomValueGenerator();
+  const dateStringTransformations = new DateStringTransformations();
+  const idToken = valueGenerator.string();
+
+  let service: DealService;
+  let acbsDealServiceCreateDeal: jest.Mock;
+  let currentDateProviderGetEarliestDateFromTodayAnd: jest.Mock;
+
+  beforeEach(() => {
+    const acbsDealService = new AcbsDealService(null, null);
+    acbsDealServiceCreateDeal = jest.fn();
+    acbsDealService.createDeal = acbsDealServiceCreateDeal;
+
+    const acbsAuthenticationService = new AcbsAuthenticationService(null, null, null);
+    const acbsAuthenticationServiceGetIdToken = jest.fn();
+    acbsAuthenticationService.getIdToken = acbsAuthenticationServiceGetIdToken;
+    when(acbsAuthenticationServiceGetIdToken).calledWith().mockResolvedValueOnce(idToken);
+
+    const currentDateProvider = new CurrentDateProvider();
+    currentDateProviderGetEarliestDateFromTodayAnd = jest.fn();
+    currentDateProvider.getEarliestDateFromTodayAnd = currentDateProviderGetEarliestDateFromTodayAnd;
+
+    service = new DealService(acbsAuthenticationService, acbsDealService, dateStringTransformations, currentDateProvider);
+  });
+
+  describe('createDeal', () => {
+    const portfolioIdentifier = PROPERTIES.GLOBAL.portfolioIdentifier;
+    const dealIdentifier = valueGenerator.stringOfNumericCharacters({ length: 10 });
+    const currency = TEST_CURRENCIES.A_TEST_CURRENCY;
+    const dealValue = 123.45;
+    const guaranteeCommencementDateAsDate = new Date('2019-01-02T00:00:00Z');
+    const guaranteeCommencementDateInPast = dateStringTransformations.getDateOnlyStringFromDate(guaranteeCommencementDateAsDate);
+    const guaranteeCommencementDateString = dateStringTransformations.getDateStringFromDate(guaranteeCommencementDateAsDate);
+    const guaranteeCommencementDateForDescription = '02/01/2019';
+    const now = new Date();
+    const midnightToday = dateStringTransformations.getDateStringFromDate(now);
+    const todayFormattedForDescription = dateStringTransformations.getDateOnlyStringFromDate(now).split('-').reverse().join('/');
+    const obligorPartyIdentifier = valueGenerator.stringOfNumericCharacters({ length: 8 });
+    const obligorName = valueGenerator.string({ maxLength: 19 });
+    const obligorIndustryClassification = valueGenerator.string({ maxLength: 10 });
+
+    const dealToCreate = {
+      dealIdentifier,
+      currency,
+      dealValue,
+      guaranteeCommencementDate: guaranteeCommencementDateInPast,
+      obligorPartyIdentifier: obligorPartyIdentifier,
+      obligorName: obligorName,
+      obligorIndustryClassification: obligorIndustryClassification,
+    };
+
+    const getExpectedDescription = ({ obligorName, currency, formattedDate }: { obligorName: string; currency: string; formattedDate: string }): string =>
+      `D: ${obligorName} ${currency} ${formattedDate}`;
+
+    const defaultValues = PROPERTIES.DEAL.DEFAULTS;
+    const expectedDealToCreateInAcbs = {
```

Comment:
I could move this out of this file, and out of `post-deal.api-test.ts`, because it's repeated and takes up a lot of space